### PR TITLE
Update printer.go for used/not-used ETCD storage percentage

### PR DESCRIPTION
**Pull request by Scotchman0**
based on OMC's output for etcdctl endpoint health reporting, add output that includes how much storage is used/not in use and a percentage for better visualization of data within the database. Help with analysis for defrag rates and visual output.

https://github.com/gmeghnag/omc/blob/e97cc10586cefbf0c626ef43b61597730f2b82cf/cmd/etcd/etcdctl.go#L35-L50\

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

**Comment by Scotchman0** [comment]
Example output:

~~~
+-----------------------------+------------------+---------+----------------+----------+-----------+------------+-----------+------------+--------------------+--------+
|          ENDPOINT           |        ID        | VERSION | DB SIZE/IN USE | NOT USED | IS LEADER | IS LEARNER | RAFT TERM | RAFT INDEX | RAFT APPLIED INDEX | ERRORS |
+-----------------------------+------------------+---------+----------------+----------+-----------+------------+-----------+------------+--------------------+--------+
| https://xxx.xxx.xxx:2379  | 1fc63d8d0eb5fa7b | 3.5.9   | 6.7 GB/5.6 GB  | 18%      | false     | false      |      5853 | 4073918326 |         4073918326 |        |
| https://xxx.xxx.xxx:2379 | 3bd2d35c028a6a2f | 3.5.9   | 6.8 GB/5.6 GB  | 19%      | true      | false      |      5853 | 4073918327 |         4073918327 |        |
| https://xxx.xxx.xxx:2379 | 41cbbc084d60007d | 3.5.9   | 6.8 GB/5.6 GB  | 18%      | false     | false      |      5853 | 4073918330 |         4073918330 |        |
+-----------------------------+------------------+---------+----------------+----------+-----------+------------+-----------+------------+--------------------+--------+
~~~

**Comment by ivanvc** [comment]
Hi @Scotchman0, thanks for your contribution. While we await a maintainer to review this suggestion, would you mind signing your commit? So the developer certificate of origin (DCO) check passes, i.e:

```
git rebase HEAD~1 --signoff
git push --force
```

Ref: https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#commit-your-change

**Comment by Scotchman0** [comment]
Latest update:

~~~
+-----------------------------+------------------+---------+----------------+----------+-----------+------------+-----------+------------+--------------------+--------+
|          ENDPOINT           |        ID        | VERSION | DB SIZE | IN USE | NOT USED | QUOTA | IS LEADER | IS LEARNER | RAFT TERM | RAFT INDEX | RAFT APPLIED INDEX | ERRORS |
+-----------------------------+------------------+---------+----------------+----------+-----------+------------+-----------+------------+--------------------+--------+
| https://xxx.xxx.xxx:2379    | 1fc63d8d0eb5fa7b | 3.5.9   | 6.7 GB | 5.6 GB  | 18%      | ??    | false     | false      |      5853 | 4073918326 |         4073918326 |        |
| https://xxx.xxx.xxx:2379    | 3bd2d35c028a6a2f | 3.5.9   | 6.8 GB | 5.6 GB  | 19%      | ??    | true      | false      |      5853 | 4073918327 |         4073918327 |        |
| https://xxx.xxx.xxx:2379    | 41cbbc084d60007d | 3.5.9   | 6.8 GB | 5.6 GB  | 18%      | ??    | false     | false      |      5853 | 4073918330 |         4073918330 |        |
+-----------------------------+------------------+---------+----------------+----------+-----------+------------+-----------+------------+--------------------+---------
~~~

File: etcdctl/ctlv3/command/printer.go
Comment (Scotchman0) [reply etcdctl/ctlv3/command/printer.go]: changed - agreed that's clearer

File: etcdctl/ctlv3/command/printer.go
Comment (Scotchman0) [reply etcdctl/ctlv3/command/printer.go]: ack - would the following format handling also succeed here? - pushed to check it but I can also revisit how I'm calculating the percentage with more of a robust option if it doesn't like this:

`humanize.Bytes(uint64(100-(status.Resp.DBsizeInUse*100/status.Resp.DBSize))) + "%",`

File: etcdctl/ctlv3/command/printer.go
Comment (Scotchman0) [inline etcdctl/ctlv3/command/printer.go]: removed this for the time being to see about clearing the checks but will want to add it in before merge pending the other PR that will be setting this value - will revisit.

File: etcdctl/ctlv3/command/printer.go
Comment (Scotchman0) [reply etcdctl/ctlv3/command/printer.go]: looks like no. :thinking: I'll explore some alternatives

File: etcdctl/ctlv3/command/printer.go
Comment (siyuanfoundation) [reply etcdctl/ctlv3/command/printer.go]: can you run the build and test locally first?

**Comment by siyuanfoundation** [comment]
Is it important to have `NOT IN USE` in percentage? If it is not critical, it is better to keep it in MB. Otherwise, can you change the name to `PERCENTAGE NOT IN USE`?

**Comment by Scotchman0** [comment]
> Is it important to have `NOT IN USE` in percentage? If it is not critical, it is better to keep it in MB. Otherwise, can you change the name to `PERCENTAGE NOT IN USE`?

I'd be inclined to change the name; The goal is to add an easy marker for identification of unused disk space which might help to illustrate why the cluster defrags often or whether or not a defrag is expected / could benefit performance. Mostly this is for easy visibility into this logic: https://github.com/openshift/cluster-etcd-operator/blob/cbfb856ec8892687a303989b84e01c8f34c1967e/pkg/operator/defragcontroller/defragcontroller.go#L181-L224
And where the max value is set: https://github.com/openshift/cluster-etcd-operator/blob/cbfb856ec8892687a303989b84e01c8f34c1967e/pkg/operator/defragcontroller/defragcontroller.go#L31C2-L31C45 (45%). 

To be consistent, I think percentage handling is preferable. I've changed the name to `PERCENTAGE NOT IN USE` as requested and pushed a new build. 

Thanks!

**Comment by Scotchman0** [comment]
oof - I accidentally broke stuff while forcing DCO :facepalm: 1s I'll revert.

**Comment by Scotchman0** [comment]
re-syncing with main - @siyuanfoundation let me know if any additional changes are required; thanks!

**Review by siyuanfoundation — APPROVED** [review]

**Comment by ahrtr** [comment]
@Scotchman0 can you please provide an example output using latest `etcdserver` + `etcdctl` on main branch?

**Comment by Scotchman0** [comment]
~~~
 sunbro@apollo  ~/Downloads/etcd/bin   patch-1  ./etcdctl endpoint status -w table
+----------------+------------------+---------------+-----------------+---------+--------+-----------------------+-------+-----------+------------+-----------+------------+--------------------+--------+
|    ENDPOINT    |        ID        |    VERSION    | STORAGE VERSION | DB SIZE | IN USE | PERCENTAGE NOT IN USE | QUOTA | IS LEADER | IS LEARNER | RAFT TERM | RAFT INDEX | RAFT APPLIED INDEX | ERRORS |
+----------------+------------------+---------------+-----------------+---------+--------+-----------------------+-------+-----------+------------+-----------+------------+--------------------+--------+
| 127.0.0.1:2379 | 8e9e05c52164694d | 3.6.0-alpha.0 |           3.6.0 |   20 kB |  16 kB |                   20% | 11 GB |      true |      false |         3 |          8 |                  8 |        |
+----------------+------------------+---------------+-----------------+---------+--------+-----------------------+-------+-----------+------------+-----------+------------+--------------------+--------+

./etcdctl endpoint status -w json 
[{"Endpoint":"127.0.0.1:2379","Status":{"header":{"cluster_id":14841639068965178418,"member_id":10276657743932975437,"revision":1,"raft_term":3},"version":"3.6.0-alpha.0","dbSize":20480,"leader":10276657743932975437,"raftIndex":8,"raftTerm":3,"raftAppliedIndex":8,"dbSizeInUse":16384,"storageVersion":"3.6.0","dbSizeQuota":10737418240}}]

~~~

**Review by ahrtr — APPROVED** [review]
LGTM

Thanks for your first contribution!

**Pull request merged**
